### PR TITLE
 #160371362 Comments Notification 

### PR DESCRIPTION
**Pull request by clintfidel**
#### What does this PR do?
This ensures that a notification status is updated when notification is read

#### Description of Task to be completed?
- Update comment notification to ensure that requester get a notification when a comment is posted 

#### How should this be manually tested?
- Clone the repo - `git clone https://github.com/andela/travel_tool_back.git`
- Enter into the project directory - `cd travel_tool_back`
- Rollback and run database migrations - `yarn db:rollmigrate`
- Install dependencies and start the server - `make start`
- post a comment using postman `localhost:5000/api/v1/comments`

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?
[ #160371362](https://www.pivotaltracker.com/story/show/160371362)

#### Screenshots (if appropriate)
None

### CODE REVIEW

**Comment by clintfidel** [comment]
@albert85 , that has been corrected, please check

File: src/database/models/notifications.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+
```
Comment (albert85) [inline src/database/models/notifications.js]: Please update this models and its migration files with develop branch

**Comment by clintfidel** [comment]
@albert85 , merge conflict has been fixed

File: src/helpers/socket/newRequest.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export default function newRequest(client, io) {
+  client.on('createRequest', (data) => {
+    console.log('+++ data from newRequest connection +++ ', data, client.id);
```
Comment (hariclerry) [inline src/helpers/socket/newRequest.js]: Can you please remove this console.log() line?

File: src/helpers/Utils.js [hunk 3]
```diff
@@ -14,6 +14,22 @@ class Utils {
...
+    if (typeof data !== 'object') {
+      return;
+    }
+    const validKeys = ['senderId', 'recipientId', 'notificationType',
+      'requestId', 'message', 'notificationLink',
+      'notificationStatus', 'senderName', 'senderImage'];
+    const dataKeys = Object.keys(data);
+    validKeys.map((value) => {
+      if (dataKeys.includes(value) === false) {
+        return 'invalid data';
```
Comment (hariclerry) [inline src/helpers/Utils.js]: Can you please change the `return 'invalid data'` to  `return 'Invalid data'`. Just a grammatical error I presume.

**Comment by clintfidel** [comment]
@hariclerry , changes have been implemented, please confirm

**Comment by seunzone** [comment]
@clintfidel kindly check the failing test.

**Comment by clintfidel** [comment]
@seunzone , the failing test have been addressed on @Nedson202 PR.

**Comment by seunzone** [comment]
> @seunzone , the failing test have been addressed on @Nedson202 PR.

Thanks for clarifying that

**Comment by seunzone** [comment]
Can you drop a screenshot of any form of communication indicating that the test is been fixed on a different branch probably here on github or on your activity pane on pivotal tracker @clintfidel

**Comment by clintfidel** [comment]
@albert85 , please refer to the previous comment I dropped on seun's Feedback

File: src/helpers/Utils.js [hunk 4]
```diff
@@ -14,6 +14,22 @@ class Utils {
     return token;
   }
 
+  static validateNotificatioData(data) {
```
Comment (amarachukwu-agbo) [inline src/helpers/Utils.js]: Please fix the typo in this method name

**Pull request closed without merging**